packages/go-nogo/src/index.ts

```diff
@@ -0,0 +1,484 @@
+ * Creates a set of instructions for the Go/No-Go task.
+ * 
+ * @param {string[]} instructions - Array of instruction pages to display.
+ * @param {object} texts - Text configuration object containing messages and labels.
+ * @returns {object} jsPsychInstructions trial object.
+ */
+export function createInstructions(instructions: string[] = instruction_pages, texts = trial_text) {
+  if (instructions.length === 0) {
+    instructions = instruction_pages;
+  }
+  return {
+    type: jsPsychInstructions,
+    pages: instructions.map(page => `<div class="timeline-instructions"><p>${page}</p></div>`),
+    show_clickable_nav: true,
+    allow_keys: true,
+    key_forward: 'ArrowRight',
+    key_backward: 'ArrowLeft',
+    button_label_previous: texts?.back_button ?? texts.back_button,
+    button_label_next: texts?.next_button ?? texts.next_button,
+    data: { task: 'go-nogo', phase: 'instructions' }
+  };
+}
+
+// Helper function to create stimulus HTML
+const createStimulusHTML = (html?: string, isGoTrial?: boolean): string => {
+  const id = isGoTrial ? 'go-stimulus' : 'nogo-stimulus'
+  
+  if (html) {
+    return `<div id="${id}-container" class="go-nogo-container timeline-trial">${html}</div>`
+  }
+  
+  return `
+    <div id="${id}-container" class="go-nogo-container timeline-trial"><h1 id="${id}" class="go-nogo-stimulus">${isGoTrial ? 'Y' : 'X'}</h1></div>`
+}
+
+/**
+ * Creates a Go instruction trial with practice and infinite retry logic.
+ * 
+ * This function creates a dynamic timeline that teaches users how to respond to "go" stimuli.
+ * Users must click the button within 10 seconds to proceed. If they fail to click in time,
+ * they receive failure feedback and the trial repeats until they succeed.
+ * 
+ * @param go_stimulus - The stimulus content to display (can be text or HTML)
+ * @param texts - Text configuration object containing all messages and labels
+ * 
+ * @returns A timeline object containing:
+ *   - Practice task with 10-second timeout
+ *   - Success feedback (if clicked in time)
+ *   - Failure feedback + retry loop (if timeout occurs)
+ */
+const createGoInstructionTrial = (go_stimulus: string, texts = trial_text) => {
+  const goExample = createStimulusHTML(go_stimulus, true)
+  
+  // Create a timeline to hold the practice trials
+  // This will allow us to conditionally push trials for retry logic
+  const practiceGoTimeline = []
+  
+  const practiceTask = {
+    type: jsPsychHtmlButtonResponse,
+    stimulus: `
+      <p>${texts.goPageContent}</p>
+      ${goExample}
+      <div class="go-nogo-feedback" style="visibility: hidden;">${texts.goodJobMessage}</div>
+    `,
+    choices: [texts.defaultButtonText],
+    trial_duration: 10000, // 10 seconds timeout
+    data: { task: 'go-nogo', phase: 'go-practice' },
+    button_html: (choice, choice_index) => `<button id="go-nogo-btn" class="jspsych-btn timeline-html-btn">${choice}</button>`,
+    on_finish: (data: any) => {
+      if (data.response !== null) {
```

Comment:
should we add something like `data.correct = data.response !== null` to add a `correct` value to the data? does that happen in the main trials?